packages/react-stand-in/src/createClassProxy.js

```diff
@@ -67,6 +83,10 @@ function createClassProxy(InitialComponent, proxyKey, wrapResult = identity) {
         ? CurrentComponent(this.props, this.context)
         : CurrentComponent.prototype.render.call(this)
 
+      if (!isReactElement(result)) {
```

Comment:
The actual fix.

Reply:
I'm not sure this is right. Per https://github.com/facebook/react/blob/e6e393b9c5221bfb1a5ddcc7221c42e96ab3baca/packages/react-reconciler/src/ReactFiberBeginWork.js#L446-L450, the "indeterminate component" logic seems to only get run for the initial mount.

This logic will actually construct a new component instance on every render call, which doesn't seem like what we want.

We shouldn't be creating new class instances if we can help it at all.

Reply:
Maybe the answer is to cache `result` in this case? It shouldn't invalidate, I don't think.

Reply:
The logic is the simple - `how it will work __without__ RHL?`
Without RHL it will construct and return a new Component on each render call, and will not _cache_ anything.
That's the answer.

We can do a lot, but we shouldn't.

Reply:
Are you sure about that? The snippet above has this check:

```js
    invariant(
      current === null,
      'An indeterminate component should never have mounted. This error is ' +
        'likely caused by a bug in React. Please file an issue.',
    );
```

So I believe this logic only gets applied at initial mount?

Reply:
There is no such thing as `initial` mount for a function stateless component. As I understand - it just can't `change its mind`.
In real - this code works without any exceptions or warnings.

Reply:
Random question: why can't we just wrap functional components separately (not as a react component), and then _only_ create a proxy for it's results if it generates a valid react component? 
If it's returning an element or anything else we don't really need to care. This would also be much simpler, and probably add less overhead...

Reply:
Cos first you have to wrap them, next you can test something.